File: 0283.移动零/solution.go
package main

func moveZeroes(nums []int) {
    slow, fast := 0, 0
    for fast <= len(nums)-1 {
        if nums[fast] == 0 {
            fast++
        } else {
            nums[slow], nums[fast] = nums[fast], nums[slow]
            slow++
            fast++
        }
    }
}

File: 0283.移动零/Solution.java
class Solution {
    public void moveZeroes(int[] nums) {
        int slow = 0;
        int fast = 0;
        while (fast <= nums.length - 1) {
            if (nums[fast] == 0) {
                fast++;
            } else {
                int tmp = nums[slow];
                nums[slow] = nums[fast];
                nums[fast] = tmp;
                fast++;
                slow++;
            }
        }
    }
}